html_app/build.py

```diff
@@ -0,0 +1,257 @@
+    if path.endswith(".soy"):
+        infile = path
+        outfile = "{}/gen/{}.js".format(
+            os.path.dirname(infile),
+            os.path.basename(infile)
+        )
+        call([
+            "java",
+            "-jar", "../tools/SoyToJsSrcCompiler.jar",
+            "--outputPathFormat", outfile,
+            infile
+        ])
+        print "compile soy {} ".format(path)
+    if path.endswith(".gss"):
+        infile = path
+        outfile = "{}/gen/{}.css".format(
+            os.path.dirname(infile),
+            os.path.basename(infile)
+        )
+        call([
+            "java",
+            "-jar", "../tools/closure-stylesheets-20111230.jar",
+            "--pretty-print", infile,
+            "-o", outfile
+        ])
+        print "compile gss {} to {} ".format(infile, outfile)
+    if path.endswith(".touch_index"):
+        update_index_html()
+    if update_index:
+        global_update_index = True
+        os.utime("{}.touch_index".format(ROOT), None)
+
+
+def process_all():
+    for subdir, _, files in os.walk(ROOT):
+        for f in files:
+            path = "{}/{}".format(subdir, f)
+            processor(path)
+
+    update_index_html()
+
+
+class StaticConversionEventHandler(RegexMatchingEventHandler):
+    """
+    Match static files in the source directory and translate them.
+    """
+
+    def __init__(self):
+        super(StaticConversionEventHandler, self).__init__(
+            regexes=[".*\.(soy|gss)$"],
+            ignore_directories=True,
+            case_sensitive=False
+        )
+
+    def do_conversion(self, event):
+        """
+        Translate the static assets.
+        """
+        processor(event.src_path)
+        update_index_html()
+
+    def on_modified(self, event):
+        """
+        Process modified markdown files.
+        """
+        self.do_conversion(event)
+
+    def on_created(self, event):
+        """
+        Process newly created markdown files.
+        """
+        self.do_conversion(event)
+
+
+def watch(source):
+    """
+    Watch the source directory for changes until a KeyboardInterrupt is
+    encountered.
+    """
+    observer = Observer()
+    event_handler = StaticConversionEventHandler()
+    observer.schedule(event_handler, source, recursive=True)
+
+    logging.info("Starting watcher..")
+
+    try:
+        observer.start()
+        while True:
+            time.sleep(1)
+    except KeyboardInterrupt:
+        logging.info("Keyboard interrupt received; stopping watcher..")
+        observer.stop()
+
+    observer.join()
+
+
+if __name__ == "__main__":
+    args = docopt(__doc__)
+
+    soy_only = args["--soy-only"]
+    gss_only = args["--gss-only"]
+
+    logging.basicConfig(
+        level=logging.INFO,
+        format='[%(asctime)s] %(levelname)s: %(message)s',
+        datefmt='%Y-%m-%d %H:%M:%S'
+    )
+
+    if args['watch']:
+        watch(".")
+    else:
+        # Run a build
+        if soy_only or gss_only:
```

Comment:
I'd generally prefer not to add options that aren't implemented.  Are you planning to implement these soon?